drivers/sensor/lsm6dsl/Kconfig

```diff
@@ -8,41 +8,93 @@
 #
 
 menuconfig LSM6DSL
-	bool "LSM6DSL I2C accelerometer and gyroscope Chip"
-	depends on I2C
+	bool "LSM6DSL I2C/SPI accelerometer and gyroscope Chip"
+	depends on SENSOR && (I2C || SPI)
 	default n
 	help
-	  Enable driver for LSM6DSL I2C-based accelerometer and gyroscope
+	  Enable driver for LSM6DSL accelerometer and gyroscope
 	  sensor.
 
-if !HAS_DTS_I2C_DEVICE
-
 config LSM6DSL_DEV_NAME
 	string "LSM6DSL device name"
 	depends on LSM6DSL
 	default "LSM6DSL"
 
+choice
+	prompt "Interface to AP"
+	depends on LSM6DSL
+	default LSM6DSL_SPI
+	help
+	  Select interface the LSM6DSL driver is connected to AP
+
+config LSM6DSL_SPI
+	depends on SPI
+	bool "SPI Interface"
+
+config LSM6DSL_I2C
+	depends on I2C
+	bool "I2C Interface"
+
+endchoice
+
 config LSM6DSL_I2C_ADDR
 	hex
 	prompt "LSM6DSL I2C address"
-	depends on LSM6DSL
+	depends on LSM6DSL && LSM6DSL_I2C
```

Comment:
Is LSM6DSL dependency actually needed here? (same question everywhere)

Reply:
Maybe not, but it is there everywhere...